test/conftest.py

```diff
@@ -91,36 +107,75 @@ def test_workspace(
     :type config: dict[str, str]
     :param context: context object
     :type context: dict[str, str|list]
-    :yield: dict containing ws name, ws ID, and a workspace client.
+    :yield: dict containing a workspace client, and ws name and ws ID for the default and list all sets workspaces.
     :rtype: dict[str, Any]
     """
     ws_name = f"test_SetAPI_{int(time.time() * 1000)}"
+    list_all_sets_ws_name = f"list_all_sets_{ws_name}"
     ws_client = Workspace(config["workspace-url"], token=TOKEN)
-    # create a workspace
-    result = ws_client.create_workspace({"workspace": ws_name})
-    yield {"ws_client": ws_client, "ws_name": ws_name, "ws_id": result[0]}
-    # delete the test workspace
-    deletion = ws_client.delete_workspace({"workspace": ws_name})
+    # create workspaces for the tests
+    default_ws_info = ws_client.create_workspace({"workspace": ws_name})
+    list_all_sets_ws_info = ws_client.create_workspace(
+        {"workspace": list_all_sets_ws_name}
+    )
+
+    ws_id = default_ws_info[0]
+    list_all_sets_ws_id = list_all_sets_ws_info[0]
+    yield {
+        "ws_client": ws_client,
+        "ws_name": ws_name,
+        "ws_id": ws_id,
+        "list_all_sets_ws_name": list_all_sets_ws_name,
+        "list_all_sets_ws_id": list_all_sets_ws_id,
```

Comment:
these will be used for the `list_all_sets` tests, not surprisingly! Since that test starts with a workspace bereft of sets, it has to be done with a new ws, not the one used for all the other tests.